src/Scrutor/AttributeSelector.cs

```diff
@@ -40,30 +40,42 @@ void ISelector.Populate(IServiceCollection services)
 
                 foreach (var attribute in attributes)
                 {
-                    var serviceType = GetServiceType(type, attribute);
+                    var serviceTypes = GetServiceTypes(type, attribute);
 
-                    var descriptor = new ServiceDescriptor(serviceType, type, attribute.Lifetime);
+                    foreach (var serviceType in serviceTypes)
+                    {
+                        var descriptor = new ServiceDescriptor(serviceType, type, attribute.Lifetime);
 
-                    services.Add(descriptor);
+                        services.Add(descriptor);
+                    }
                 }
             }
         }
 
-        private static Type GetServiceType(Type type, ServiceDescriptorAttribute attribute)
+        private static IEnumerable<Type> GetServiceTypes(Type type, ServiceDescriptorAttribute attribute)
         {
             var serviceType = attribute.ServiceType;
 
             if (serviceType == null)
             {
-                return type;
+                var typeInfo = type.GetTypeInfo();
+                yield return type;
```

Comment:
Yep. Looks good. I'll have a closer look later this weekend and push a new version. Thanks :grin: